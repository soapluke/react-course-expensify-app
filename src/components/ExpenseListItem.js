import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}><h3>Description: {description}</h3></Link>
        <p>Amount: {amount}</p>
        <p>Created At: {moment(createdAt).format('MMMM Do, YYYY')}</p>
    </div>
);

export default ExpenseListItem;