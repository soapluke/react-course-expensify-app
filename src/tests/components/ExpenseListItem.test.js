import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem key={expenses[1].id} {...expenses[1]}/>);

    expect(toJSON(wrapper)).toMatchSnapshot();
});