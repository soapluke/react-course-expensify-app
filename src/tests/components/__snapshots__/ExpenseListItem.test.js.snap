// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Description: 
      Rent
    </h3>
  </Link>
  <p>
    Amount: 
    109500
  </p>
  <p>
    Created At: 
    -345600000
  </p>
</div>
`;
